src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -41,45 +37,54 @@ public void RelativeSearchPath_Is_Null()
             Assert.Null(AppDomain.CurrentDomain.RelativeSearchPath);
         } 
 
-
         [Fact]
         public void UnhandledException_Add_Remove()
         {
-            AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(MyHandler);
-            AppDomain.CurrentDomain.UnhandledException -= new UnhandledExceptionEventHandler(MyHandler);
+            RemoteInvoke(() => {
+                AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(MyHandler);
+                AppDomain.CurrentDomain.UnhandledException -= new UnhandledExceptionEventHandler(MyHandler);
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void UnhandledException_NotCalled_When_Handled()
         {
-            AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(NotExpectedToBeCalledHandler);
-            try {
-                throw new Exception();
-            }
-            catch
-            {
-            }
-            AppDomain.CurrentDomain.UnhandledException -= new UnhandledExceptionEventHandler(NotExpectedToBeCalledHandler);
+            RemoteInvoke(() => {
+                AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(NotExpectedToBeCalledHandler);
+                try
+                {
+                    throw new Exception();
+                }
+                catch
+                {
+                }
+                AppDomain.CurrentDomain.UnhandledException -= new UnhandledExceptionEventHandler(NotExpectedToBeCalledHandler);
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [ActiveIssue(12716)]
         [PlatformSpecific(~TestPlatforms.OSX)] // Unhandled exception on a separate process causes xunit to crash on osx
         [Fact]
         public void UnhandledException_Called()
         {
-            System.IO.File.Delete("success.txt");
-            RemoteInvokeOptions options = new RemoteInvokeOptions();
-            options.CheckExitCode = false;
-            RemoteInvoke(() =>
-            {
-                AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(MyHandler);
-                throw new Exception("****This Unhandled Exception is Expected****");
+            RemoteInvoke(() => {
+                System.IO.File.Delete("success.txt");
```

Comment:
Doesn't this fail in app container? It should be in %temp%, eg by using System.IO.TempFile from src\Common\tests\System\IO\TempFile.cs